speaker/speaker.go

```diff
@@ -25,57 +26,40 @@ var (
 // bufferSize means lower CPU usage and more reliable playback. Lower bufferSize means better
 // responsiveness and less delay.
 func Init(sampleRate beep.SampleRate, bufferSize int) error {
-	mu.Lock()
-	defer mu.Unlock()
-
-	Close()
+	if context != nil {
+		return errors.New("speaker cannot be initialized more than once")
+	}
 
 	mixer = beep.Mixer{}
 
-	numBytes := bufferSize * 4
-	samples = make([][2]float64, bufferSize)
-	buf = make([]byte, numBytes)
-
 	var err error
-	context, err = oto.NewContext(int(sampleRate), 2, 2, numBytes)
+	var readyChan chan struct{}
+	context, readyChan, err = oto.NewContext(int(sampleRate), channelCount, bitDepthInBytes)
 	if err != nil {
 		return errors.Wrap(err, "failed to initialize speaker")
 	}
-	player = context.NewPlayer()
-
-	done = make(chan struct{})
+	<-readyChan
 
-	go func() {
-		for {
-			select {
-			default:
-				update()
-			case <-done:
-				return
-			}
-		}
-	}()
+	player = context.NewPlayer(newReaderFromStreamer(&mixer))
+	player.(oto.BufferSizeSetter).SetBufferSize(bufferSize * bytesPerSample)
+	player.Play()
 
 	return nil
 }
 
-// Close closes the playback and the driver. In most cases, there is certainly no need to call Close
-// even when the program doesn't play anymore, because in properly set systems, the default mixer
-// handles multiple concurrent processes. It's only when the default device is not a virtual but hardware
-// device, that you'll probably want to manually manage the device from your application.
+// Close closes audio playback. However, the underlying driver context keeps existing, because
+// closing it isn't supported (https://github.com/hajimehoshi/oto/issues/149). In most cases,
+// there is certainly no need to call Close even when the program doesn't play anymore, because
+// in properly set systems, the default mixer handles multiple concurrent processes.
 func Close() {
 	if player != nil {
-		if done != nil {
-			done <- struct{}{}
-			done = nil
-		}
 		player.Close()
-		context.Close()
```

Comment:
So closing the context isn't supported by Oto. I think this is the only breaking change.

Reply:
Yeah, is this problematic?